proj/src/main/webapp/script.js

```diff
@@ -27,11 +27,72 @@ if(btnLogin){
     });
 }
 
+
+// Adds user to an existing chat when given a reference to the place in the database
+function addUserToTag(reference){
+    var currentUID = firebase.auth().currentUser.uid
+    console.log("adding new user to chat room with uid: " + currentUID);
+    reference.push(currentUID);
+}
+
+// Creates a new chat given a tag and adds the current user as a member
+function createNewChatWithUser(tag){
+    console.log("creating new chat with tag: " + tag);
+    var newChat = {
+        "name" : tag,
+        "tag" : tag,
+    };
+    var currentReference = firebase.database().ref("/chat/" + tag);
+    var postKey = currentReference.push(newChat).key;
+    currentReference = firebase.database().ref("/chat/" + tag + "/" + postKey + "/users/");
+    addUserToTag(currentReference);
+}
+
+// Create or join chatroom
+function createOrJoinChat(currentTag){
+    var ref = firebase.database().ref("/chat/");
+    ref.once("value").then(function(snapshot){
+        // Checks to see if tag already exists in database
+        if(snapshot.hasChild(currentTag)){
+
+            // If tag already exists, navigate to users and add users if there is room
+            var query = firebase.database().ref("/chat/" + currentTag + "/").orderByKey();
+            query.once("value").then(function(snapshot){
+
+                snapshot.forEach(function(childSnapshot){
+
+                    var currentReference = childSnapshot.ref;
+                    if(childSnapshot.hasChild("users")){
+
+                        var userSnapshot = childSnapshot.child("users");
+                        var usersReference = userSnapshot.ref;
+                        if(userSnapshot.numChildren() < 200){
+                            addUserToTag(usersReference);
+                            return;
```

Comment:
I see, return if you've added the user, otherwise you will finish cycling through the list of chats and create a new chat below.